test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -1347,21 +1385,55 @@ void ProtoConverter::visit(PopStmt const& _x)
 	m_output << ")\n";
 }
 
+void ProtoConverter::visit(Code const& _x)
+{
+	m_output << "code {\n";
+	visit(_x.block());
+	m_output << "}\n";
+}
+
+void ProtoConverter::visit(Data const& _x)
+{
+	m_output << "data \"datablock\" hex\"" << createHex(_x.hex()) << "\"\n";
```

Comment:
Shouldn't this use a variable id?

Reply:
I followed the yul specification here https://solidity.readthedocs.io/en/v0.5.11/yul.html?highlight=dataoffset#specification-of-yul-object

that states

```
Data = 'data' StringLiteral ( HexLiteral | StringLiteral )
```

Do you mean extending this to something like so?

```
Data = 'data' StringLiteral ( HexLiteral | StringLiteral | VarRef)
```

Reply:
I meant that it always uses `"datablock"`

Reply:
Ah I see. So `datasize` and `dataoffset` may either invoke an object or a data ID, did I get that right?

Reply:
Right now, the first arguments of `datasize` and `dataoffset` reference the identifier of the currently "visited" object.

Reply:
Ah I see! Yes, you can reference the current object, a sub-object or a data object in the current object.

Reply:
Sounds good. I created a separate issue for the sub-object thing (see #7381 ). I will update this PR so that datasize and dataoffset can reference either the current object or data block.

Then, I can generalize the behavior for object referencing as part of #7381